test/integration/silent-payments.spec.ts

```diff
@@ -0,0 +1,181 @@
+
+import { regtestUtils } from './_regtest';
+import * as bitcoin from '../..';
+import { toXOnly } from '../../src/psbt/bip371';
+
+import { tweakSigner } from './taproot.utils';
+
+const rng = require('randombytes');
+const regtest = regtestUtils.network;
+bitcoin.initEccLib(ecc);
+const bip32 = BIP32Factory(ecc);
+
+describe('bitcoinjs-lib (silent payments)', () => {
+  // for simplicity the transactions in this test have only one input and one output
+  it('can create (and broadcast via 3PBP) a simple silent payment', async () => {
+    const { senderKeyPair, receiverKeyPair, sharedSecret } = initParticipants();
+
+    // this is what the sender sees/scans (from twitter bio, public forum, truck door)
+    const silentPublicKey = toXOnly(receiverKeyPair.publicKey);
+
+    const senderUtxo = await fundP2pkhUtxo(senderKeyPair.publicKey);
+    // amount to pay the silent address
+    const payAmount = senderUtxo.value - 1e4;
+
+    // The sender pays to the tweaked slient adddress
+    const {
+      psbt: payPsbt,
+      address: tweakedSilentAddress,
+    } = buildPayToSilentAddress(
+      senderUtxo.txId,
+      senderUtxo,
+      silentPublicKey,
+      payAmount,
+      sharedSecret,
+    );
+    payPsbt.signInput(0, senderKeyPair).finalizeAllInputs();
+
+    // the transaction paying to the silent address
+    const payTx = payPsbt.extractTransaction();
+    await broadcastAndVerifyTx(payTx, tweakedSilentAddress!, payAmount);
+
+    // the utxo with the tweaked silent address
+    const receiverUtxo = { value: payAmount, script: payTx.outs[0].script };
+    // the amount the receiver will spend
+    const sendAmount = payAmount - 1e4;
+
+    // the receiver spends from the tweaked silent address
+    const { psbt: spendPsbt, address } = buildSpendFromSilentAddress(
+      payTx.getId(),
+      receiverUtxo,
+      silentPublicKey,
+      sendAmount,
+      sharedSecret,
+    );
+
+    const tweakedSigner = tweakSigner(receiverKeyPair!, {
+      tweakHash: sharedSecret,
+      network: regtest,
+    });
+    spendPsbt.signInput(0, tweakedSigner).finalizeAllInputs();
+
+    // the transaction spending from the silent address
+    const spendTx = spendPsbt.extractTransaction();
+    await broadcastAndVerifyTx(spendTx, address!, sendAmount);
+  });
+});
+
+async function fundP2pkhUtxo(
+  senderPubKey: Buffer,
+): Promise<{ value: number; script: Buffer; txId: string }> {
+  // the input being spent
+  const { output: p2wpkhOutput } = bitcoin.payments.p2wpkh({
+    pubkey: senderPubKey,
+    network: regtest,
+  });
+
+  // amount from faucet
+  const amount = 42e4;
+  // get faucet
+  const unspent = await regtestUtils.faucetComplex(p2wpkhOutput!, amount);
+
+  return { value: amount, script: p2wpkhOutput!, txId: unspent.txId };
+}
+
+async function broadcastAndVerifyTx(
+  tx: bitcoin.Transaction,
+  address: string,
+  value: number,
+): Promise<void> {
+  await regtestUtils.broadcast(tx.toBuffer().toString('hex'));
```

Comment:
nit: we have toHex()

Reply:
What does it mean??? I'm trying to realise the first aim of bitcoin!!! Free trade and opportunities for all humankind !!!